content/blog/mozilla/2018-07-03.md

```diff
@@ -0,0 +1,217 @@
+Title: Making calls to WebAssembly fast and implementing anyref
+Date: 2018-07-02 20:00
+Author: Benjamin Bouvier
+Tags: mozilla
+Slug: mozilla-2018-h1
+Lang: en
+
+Since this is the end of the first half-year, I think it is a good time to
+reflect and show some work I've been doing over the last few months, apart from
+the regular batch of random issues, security bugs, reviews and the fixing of 24
+bugs found by our [fuzzers](https://en.wikipedia.org/wiki/Fuzzing).
+
+## Bug 1319203: Make JS to WebAssembly calls *blazingly* fast
+
+If we want more WebAssembly (wasm) adoption, there shouldn't be a big costly
+barrier between the two universes. That is, calls from one world to the other
+should be fast. For a very long time, calls from JS to asm.js/WebAssembly have
+been quite slow in Firefox. In fact, we didn't optimize them at all. For ease
+and speed of implementation at the time, asm.js call activations (data
```

Comment:
this sentence is a bit long. perhaps "For ease and speed of implementation at the time, asm.js call activations (data structures recording information about the function being currently called in the VM) were very different from the JS ones. This difference indicated some significant structural differences, like the capability to reconstruct call stack information used by `Error()` stack frames, or just tracing the stack for
garbage collection purposes. "